lib/rspec/support/object_inspector.rb

```diff
@@ -3,23 +3,51 @@ module Support
     # Provide additional output details beyond what `inspect` provides when
     # printing Time, DateTime, or BigDecimal
     module ObjectInspector
+      DEFAULT_OPTIONS = {
+        :nested => false,
+        :joiner => ',',
+        :braces => true,
+        :trailing_comma => false
+      }
       # @api private
-      def self.inspect(object)
-        if Time === object
+      # rubocop:disable CyclomaticComplexity
+      # rubocop:disable MethodLength
+      def self.inspect(object, options=DEFAULT_OPTIONS)
+        case object
+        when Time
           format_time(object)
-        elsif defined?(DateTime) && DateTime === object
-          format_date_time(object)
-        elsif defined?(BigDecimal) && BigDecimal === object
-          "#{object.to_s 'F'} (#{object.inspect})"
-        elsif RSpec::Support.is_a_matcher?(object) && object.respond_to?(:description)
-          object.description
+        when Hash
+          format_hash(object, options)
```

Comment:
> To be perfectly honest, part of my reasoning for removing Arrays from this branch was de-scoping. I just wanted to have to stop monkey patching BigDecimal when working inside of some tests :P, and I didn't expect it this pull request to last as long as it has. I want to see this through to completion, I just find myself with less and less time to work on it at the moment (and my desire for the original feature also waning).

That's good feedback.  Sorry for turning this into more of a marathon than you intended :(.  As a contributor to open source I've always appreciated the chance maintainers provide me to address feedback but not everyone has the time or energy to do that.

Let's punt on arrays for now.  Before releasing this I'd at least like for us to move away from the options hash to having separate inspectors (one for differ, one for other callers).  Do you want to take that on before we merge?  Or should we merge this as-is and then I (or one of the other @rspec/rspec) can work on refactoring it?